app/policies/user_policy.rb

```diff
@@ -0,0 +1,40 @@
+class UserPolicy < ApplicationPolicy
+  class Scope < Scope
+    def resolve
+      scope.joins(:memberships).where(memberships: { organisation: user.organisations })
+    end
+  end
+
+  def show?
+    is_user || user_belongs_to_same_organisation
+  end
+
+  def create?
```

Comment:
Yes. Users can only be created by admins of an Organisation for that Organisation.